### PR TITLE
[[FIX]] Regular args can come after args with default

### LINKED ISSUE #1779: Rest parameters and default parameters
JSHint reports `Regular parameters cannot come after default parameters` if you use both default parameter and rest parameter, but I think this is a valid use case:
```js
function f (x = 0, ...y) {}
```

### PR DESCRIPTION
**Pull request by lukeapage**
Fixes #1779 

Not a breaking change since it used to be an error, which could not be overridden, this makes it a warning which can be overridden.

Other option is to remove the warning completely.

### CODE REVIEW

**Comment by lukeapage** [comment]
rebased on master

File: tests/unit/core.js [hunk 1]
```diff
@@ -1438,7 +1438,7 @@ exports.testDefaultArguments = function (test) {
     .addError(14, "'bar' is not defined.")
     .addError(14, "'num3' was used before it was declared, which is illegal for 'param' variables.")
     .addError(15, "'num4' was used before it was declared, which is illegal for 'param' variables.")
-    .addError(18, "Regular parameters cannot come after default parameters.")
+    .addError(11, "Regular parameters should not come after default parameters.")
```
Comment (nicolo-ribaudo) [inline tests/unit/core.js]: Why did you change the line number?
```js
/* Line 11 */ testUseOfVariables: function(num = foo, // Ok, defined above
                             num1 = num, // Ok, previous argument
                             num2 = num3, // warn, next argument
                             num3 = bar, // warn, not defined
                             num4 = num4) { // warn, self reference
},
/* Line 18 */ testBadDefault: function(num, num2=1, num3) {
  return num === num2;
},
```

Comment (lukeapage) [inline tests/unit/core.js]: I have no idea how that happened - the tests fail as well because of this. I've pushed a commit to fix it. Thanks.

**Comment by jugglinmike** [comment]
Squashed and merged to `master` at commit f2a59f1ea8b31491f9dcc2c5ded32d8ca0024d74. Thanks, Luke!
